commit ae48cc92e6afc8421ad73ca580cdfc8af104d46a
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2019-04-17T00:10:59+00:00 (6 months ago)
    
    MANTA-4222 want to use "region" attribute in aperture conditions
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>
    Reviewed by: Trent Mick <trent.mick@joyent.com>
    Approved by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/docs/internal/acuserguide.md b/docs/internal/acuserguide.md
index 6a287fe..4df4151 100644
--- a/docs/internal/acuserguide.md
+++ b/docs/internal/acuserguide.md
@@ -599,6 +599,7 @@ A list of context available to include in rules
 | sourceip | ip | source ip address of the caller |
 | time | time | time of day the request was made |
 | user-agent | string | user agent of the caller |
+| region | string | manta region to which the request was made |
 | ... | ... | ... |
 
 #### CloudAPI
diff --git a/etc/config.coal.json b/etc/config.coal.json
index dff7e7d..63a953c 100644
--- a/etc/config.coal.json
+++ b/etc/config.coal.json
@@ -134,6 +134,7 @@
         "clientTimeout": 120000
     },
     "datacenter": "coal.joyent.us",
+    "region": "coal",
     "server_uuid": "f5c4d446-7639-11e7-8ded-c35deca5b8cc",
     "zone_uuid": "dbb27e00-7639-11e7-bdd5-53e6dcb96c7f"
 }
diff --git a/lib/auth.js b/lib/auth.js
index cca494b..0bf6b15 100644
--- a/lib/auth.js
+++ b/lib/auth.js
@@ -986,6 +986,7 @@ function gatherContext(req, res, next) {
     conditions.date = new Date(req._time);
     conditions.day = new Date(req._time);
     conditions.time = new Date(req._time);
+    conditions.region = req.config.region;
     var ip = req.headers['x-forwarded-for'];
     if (ip) {
         conditions.sourceip = ip.split(',')[0].trim();
diff --git a/lib/configure.js b/lib/configure.js
index ee90302..5948c44 100644
--- a/lib/configure.js
+++ b/lib/configure.js
@@ -48,6 +48,10 @@ function configure(appName, opts, dtProbes) {
     assert.object(cfg.sharkConfig, 'cfg.sharkConfig');
     assert.optionalObject(cfg.storage, 'cfg.storage');
 
+    /* Used by artedi and RBAC */
+    assert.string(cfg.datacenter, 'cfg.datacenter');
+    assert.string(cfg.region, 'cfg.region');
+
     cfg.insecurePort = opts.insecure_port || cfg.insecurePort;
     cfg.port = opts.port || cfg.port;
     cfg.name = 'Manta';
diff --git a/sapi_manifests/muskie/template b/sapi_manifests/muskie/template
index 1d5827f..03beef2 100644
--- a/sapi_manifests/muskie/template
+++ b/sapi_manifests/muskie/template
@@ -235,6 +235,7 @@
       "clientTimeout": 120000
   },
   "datacenter": "{{DATACENTER}}",
+  "region": "{{REGION}}",
   "server_uuid": "{{auto.SERVER_UUID}}",
   "zone_uuid": "{{auto.ZONENAME}}"
 
